platforms/ide/problems-api/src/main/java/org/gradle/api/problems/ProblemSpec.java

```diff
@@ -66,6 +66,18 @@ public interface ProblemSpec {
      * <p>
      * The categorization depends on the domain and don't have any constraints. Clients (i.e. IDEs) receiving problems should use the category information for
      * properly group and sort the received instances.
+     * However, we recommend to use the same conventions as the problems emitted from Gradle core use.
+     * <ul>
+     *     <li>Entries should be all-lowercase using a dash for separator (i.e. kebab-case)</li>
+     *     <li>Should be strictly hierarchical: the category declares the domain and subcategories provide further refinement</li>
+     * </ul>
+     * A few examples with a path-like notation (i.e. {@code category:subcategory1:subcategory2}).
+     * <ul>
+     *     <li>compilation:groovy-dsl</li>
+     *     <li>compilation:java:unused-import</li>
+     *     <li>deprecation:user-code-direct</li>
+     *     <li>task-selection:no-matches</li>
```

Comment:
:x: For me, those categories are already problematic since they don't convey whether this is a verification error or not. Is there some guidance how folks should create their categories?

Reply:
That's a valid point, thank for mentioning that. We had a couple of discussions already about verification failures. The plan to address this separately in the next release cycle. We already discussed it internally and our conclusion is that we should allow specifying the verification/non-verification type of the problem report in `ProblemSpec`, independently from the categorisation.

Reply:
Interesting. I was somewhat expecting that the category would have this as a top-level element, since problems fall into these buckets.